### PR TITLE
Fixed the stability check when looking for updates for an extension

### PR DESCRIPTION
**Pull request by johansundell**
Pull Request for Issue # .

### Summary of Changes
As far as I know it is not possible to import class constants, but when looking for an update to an extension Joomla was trying to do that.

I copied the const to the outside of the Updater class

### Testing Instructions
Install this extension http://pixpro.net/pixext/downloads/module/mod_helloworld/0.0.1/e8e9e3c9f3a9dd3f1d2792d83269ad64.zip and see if it has any updates.

Link to update xml http://pixpro.net/pixext/updates/module/mod_helloworld/e8e9e3c9f3a9dd3f1d2792d83269ad64.xml

### Expected result
It should have an update with version no 0.0.2

### Actual result
It has an update with version no 0.0.6

### CODE REVIEW

**Comment by mbabker** [comment]
Yes, it is possible to use class constants.  Otherwise `Joomla\CMS\Version::MAJOR_VERSION` would fail, and that doesn't.

As is, this PR is invalid because there isn't actually a bug with the import of constants, plus you are defining these constants at the namespace level and not in the class.  What is the specific bug (with error message(s) if available) you are seeing?

**Comment by johansundell** [comment]
not when you are trying to get the value with constant() function

The error is that even if my joomla installation is on "stable" it finds updates that are in "dev" mode

**Comment by mbabker** [comment]
To get a class constant you have to pass the fully qualified class name to the `constant()` function.

```php
// Works
$major = constant('Joomla\\CMS\\Version::MAJOR_VERSION');

// Does not work, will give an undefined constant message
$major = constant('MAJOR_VERSION');
```

**Comment by johansundell** [comment]
It does not on my PHP version 5.6

```
protected function stabilityTagToInteger($tag)
	{
		$constant = '\\Joomla\\CMS\\Update\\Updater::STABILITY_' . strtoupper($tag);

		if (defined($constant))
		{
			return constant($constant);
		}

		return Updater::STABILITY_STABLE;
	}
```

File: libraries/src/Updater/Adapter/ExtensionAdapter.php [hunk 1]
```diff
@@ -382,7 +382,7 @@ public function findUpdate($options)
 	 */
 	protected function stabilityTagToInteger($tag)
 	{
-		$constant = '\\Joomla\\CMS\\Update\\Updater::STABILITY_' . strtoupper($tag);
```
Comment (csthomas) [inline libraries/src/Updater/Adapter/ExtensionAdapter.php:385]: Why you can not add only one letter "r":
```php
$constant = '\\Joomla\\CMS\\Updater\\Updater::STABILITY_' . strtoupper($tag);
```

Comment (johansundell) [reply libraries/src/Updater/Adapter/ExtensionAdapter.php:385]: That was a better solution ;)

Did not know you could do that with the constant() function

**Comment by ggppdk** [comment]
I have tested this item :white_check_mark: successfully on 24ec81b00f4e05724ea0025c3f1b07d023359ded<br /><br />Thanks at @johansundell, @csthomas too for this fix,
we had to remove multiple update tags (each of different stability level) because of the bug
i was going to look into this, but did not get time for it<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18255">issues.joomla.org/tracker/joomla-cms/18255</a>.</sub>

**Comment by csthomas** [comment]
I have tested this item :white_check_mark: successfully on 24ec81b00f4e05724ea0025c3f1b07d023359ded<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18255">issues.joomla.org/tracker/joomla-cms/18255</a>.</sub>

**Comment by ghost** [comment]
RTC after two successful tests.

**Pull request merged**